### DISCUSSION TITLE
dfx killall

### DISCUSSION THREAD

**Discussion by viviveevee** [dfinity/sdk#2660] [Feature Requests]
## What problem will this feature solve?
Sometimes I’d like to easily stop my dfx server, but I have 10 projects running with different terminal different windows and I have no idea which one is running the DFX server.

## How will this feature work?
`dfx killall` or a similar command to stop all running `dfx start`ed instances on the machine

#### Request source
`icme` on the [forum](https://forum.dfinity.org/t/cant-stop-dfx-no-local-network-replica-found-nothing-to-do-but-is-still-running/8289/7)

Note to self: [internal ticket](https://dfinity.atlassian.net/browse/SDK-1077)

**Comment by viviveevee** [comment]
Possible alternative: Allow `dfx stop` to stop to influence other instances of `dfx start`

**Comment by domwoe** [comment]
I'm also having repeated issues here. I often forget to use `dfx stop` and then at some point I want to use `dfx start` again but get an error. I don't know anymore in which terminal I had used `dfx start` before, and it seems that usually only `icx-proxy` is still running. 
Hence, I'll kill `icx-proxy` and then do `dfx start` again. It would be great if this could be simpler.

**Comment by ByronBecker** [comment]
If a single dfx server is running on a single port, but multiple projects are trying to use that port, or are blocked by it - then something like:

`dfx stop <port>` 

would be a nice command to have.

Additionally, then `dfx killall` or `dfx stop --all` would just kill all instances from which `icx-proxy` is running (if on multiple ports)

**Reply by q2333gh** [reply]
```bash
btwl@btwl-virtual-machine ~/D/i/t/b/o/c2ctest (master) [255]> dfx killall
error: unrecognized subcommand 'killall'

Usage: dfx [OPTIONS] <COMMAND>

For more information, try '--help'.
btwl@btwl-virtual-machine ~/D/i/t/b/o/c2ctest (master) [2]> dfx stop --all
error: unexpected argument '--all' found

Usage: dfx stop [OPTIONS]

For more information, try '--help'.
btwl@btwl-virtual-machine ~/D/i/t/b/o/c2ctest (master) [2]> dfx --version
dfx 0.15.2
```
this didnt works in dfx 0.15.2

**Comment by bitdivine** [comment]
Request that might shape the solution a bit:
- If `dfx start` is run but the dfx pidfile is still there, check whether there is a process with that pid.  If not, delete the pidfile and carry on.  This prevents spurious "dfx is already running" errors.  I don't know why the pidfile is still there; maybe it's left behind after reboot or something like that.

**Comment by q2333gh** [comment]
I almost always cant reboot dfx with `dfx stop`and  `dfx start`  cmd. So i had to mannually `sudo reboot` the ubuntu. `dfx stop` cmd just stuck in terminal forever.

**Reply by q2333gh** [reply]
maybe i can somehow collect some log to find out  what cause `dfx stop` fail and dfx  running  when have some internal fail ? like http 503 error code or else.

**Reply by q2333gh** [reply]
by the way ,i have  almost alltime set `    export https_proxy=http://192.168.3.22:25526 
    export http_proxy=http://192.168.3.22:25526` on my ubuntu as network proxy

**Reply by q2333gh** [reply]
I tested dfx in a docker fresh new ubuntu container. and it works alright. Just my host dev machine got stop fail issue.
